### PR TITLE
Autogenerate schema element titles

### LINKED ISSUE #301: Automatically mint "title" for entities in json schema if not explicitly given
This is a continuation to changes/discussion in https://github.com/dandi/dandi-cli/pull/299#issuecomment-737522905 . That PR adds a good number of explicit `title` entries even when they could be trivially created, e.g. `title="Value Reference"` for `valueReference` at https://github.com/dandi/dandi-cli/pull/299/files#diff-295856c56542fa0dda691f493995baed932f49e94744fbacfbd7412db8c79f17R135

So we should remove hardcoded "duplicate" entries and produce title for all entries where no explicit one is defined, so all UIs could consistently display them

### PR DESCRIPTION
**Pull request by jwodder**
Closes #301.

Note that there doesn't seem to be a way to apply this to enum types.

### CODE REVIEW

File: dandi/models.py [hunk 1]
```diff
@@ -123,21 +125,28 @@ def to_dictrepr(__pydantic_cls__: Type[BaseModel]):
             .replace(__pydantic_cls__.__name__, "dict")
         )
 
+    class Config:
+        @staticmethod
+        def schema_extra(schema: Dict[str, Any]) -> None:
+            schema["title"] = name2title(schema["title"])
+            for name, prop in schema.get("properties", {}).items():
+                if prop.get("title") is None or prop["title"] == name.title():
+                    prop["title"] = name2title(name)
```
Comment (yarikoptic) [inline dandi/models.py:134]: we need a test for this.  Please add a little test which would pretty much do what `dandi/tools/pubschemata.py` does and ensure that we do get `title` set as expected in at least one record

**Review by yarikoptic — CHANGES_REQUESTED** [review]
just asking for a simple unittest.  Otherwise looks great to me, thank you @jwodder !

May be @satra could have a peek as well

File: dandi/models.py (hunk 1 above)
Comment (jwodder) [reply dandi/models.py:134]: Test added.

**Comment by satra** [comment]
LGTM - can we check that the generated json schema has the titles (especially for the enumerated types like `License`)?

**Comment by jwodder** [comment]
@satra As I said above, there doesn't seem to be a way to autogenerate titles for enum types.

File: dandi/models.py [hunk 2]
```diff
@@ -525,28 +498,20 @@ class CommonModel(DandiBaseModel):
...
-        None,
-        title="Study Target",
-        description="What the study is related to",
-        nskey="dandi",
-    )
-    license: List[LicenseType] = Field(
-        title="License", description="License of item.", nskey="schema"
+        None, description="What the study is related to", nskey="dandi"
     )
+    license: List[LicenseType] = Field(description="License of item.", nskey="schema")
```
Comment (satra) [inline dandi/models.py:503]: @jwodder - then we should not remove the title field that was added to enum types. There are a few of these. 

```
AccessType = create_enum(AccessTypeDict)
RoleType = create_enum(RoleTypeDict)
RelationType = create_enum(RelationTypeDict)
LicenseType = create_enum(LicenseTypeDict)
IdentifierType = create_enum(IdentifierTypeDict)
DigestType = create_enum(DigestTypeDict)
```

Comment (jwodder) [reply dandi/models.py:503]: Oh, you mean the enum *fields*!  No, those titles are autogenerated fine.  It's the enum entries in the `definitions` section that don't get titles.

**Review by yarikoptic — APPROVED** [review]
thank you @jwodder , looks good to me.  @satra -- if `enum` issue is "settled" , feel welcome to click Merge ;)

**Comment by yarikoptic** [comment]
Let's proceed, and if anything -- fixup after! ;)  "release is cheap" these days (and we do not need a release for this anyways) but it would avoid possible conflicts happen we decide to tune the schema

**Pull request merged**